spec/lib/trade_tariff_backend/search_client_spec.rb

```diff
@@ -1,4 +1,27 @@
 RSpec.describe TradeTariffBackend::SearchClient do
+  describe '.server_namespace' do
+    subject { described_class.server_namespace }
+
+    it { is_expected.to eql 'tariff-test' }
+
+    context 'when overridden' do
+      before do
+        @orig_namespace = described_class.server_namespace
```

Comment:
maybe we could just use let
`let(orig_namespace) { described_class.server_namespace }`
to avoid using a variable ?

Reply:
I'm not sure that would work since the let wouldn't be called until the `after` clause?

A `let!` would work but I think those are discouraged by rubocop though that may be a better approach

Reply:
Used a `let` but referenced it from the `before` to ensure its assigned prior to the value being reassigned in the `before`